src/kubernetes.cc

```diff
@@ -961,6 +961,38 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidatePodConnectivity() const {
+  try {
+    QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1");
```

Comment:
This implicitly validates the `KubernetesEndpointHost` configuration option. We could also make it use `KubernetesPodLabelSelector`.

Reply:
I'm concerned about using the pod selector, as we do this same "just check for a single thing" for nodes. I don't think this function should turn into a unit test, strictly validating that we're able to get "any" pods and nodes should suffice as a sanity check that this updater is configured properly.

Reply:
I was proposing doing the same thing as for Docker, namely adding the pod selector here to have the API check the syntax, but also overriding the limit to 1. Otherwise a user misconfiguration (a bad pod selector) could result in failing queries on every poll, which is exactly what we want to avoid.

Reply:
Good point, if we're doing this for docker we might as well do it here. It's not completely the same as how we actually invoke this later, as it's combined with the node name later on, but at least we can validate the label selector itself. Done.